tool/src/cpp/types.rs

```diff
@@ -119,7 +127,12 @@ fn gen_type_inner<W: fmt::Write>(
         }
 
         ast::TypeName::StrReference(_) => {
-            write!(out, "const {}", library_config.string_view.expr)?;
+            let maybe_const = if in_struct {
+                ""
+            } else {
+                "const "
+            };
+            write!(out, "{maybe_const}{}", library_config.string_view.expr)?;
```

Comment:
nit: just put a `write!` in each if branch?

Reply:
I personally find it cleaner to have fewer writes